crates/dekaf/src/main.rs

```diff
@@ -195,8 +196,20 @@ async fn main() -> anyhow::Result<()> {
             let acceptor = acceptor.clone();
             tokio::select! {
                 accept = kafka_listener.accept() => {
-                    let (socket, addr) = accept?;
-                    let socket = acceptor.accept(socket).await?;
+                    let (socket, addr) = match accept {
+                        Ok(res) => res,
+                        Err(e) => {
+                            tracing::error!(e=e.to_string(), "Failed to accept TCP connection");
```

Comment:
In general, my understanding is that it's best to avoid things like `e.to_string()` because it's eagerly evaluated. Tracing also has an easier shorthand for lazily evaluated `Display` rendering using for example `e=%e` or just `%e`.

Elsewhere, we've been using `error` in structured logs. IDK if there's already a standard in dekaf for `e`, but if not then I'd suggest switching to `error`.

Reply:
Ah, `%` is helpful thx. I went through a couple different iterations of trying to represent this without getting junk like `Some(Error{error: FatalAlert(UnknownCA)})`. Will keep in mind for the future